packages/web-components/src/search-widget/client/components/property-detail.svelte

```diff
@@ -0,0 +1,190 @@
+<script>
+  import { onDestroy, onMount } from 'svelte'
+  import { createEventDispatcher } from 'svelte'
+  import Fa from 'svelte-fa'
+  import { faBed, faToilet } from '@fortawesome/free-solid-svg-icons'
+  import searchWidgetStore from '../core/store'
+  import { combineAddress, getPrice, combineNumberBedTypeStyle } from '../utils/results-helpers'
+  import { INVALID_BACKGROUND_AS_BASE64 } from '../../../common/utils/constants'
+
+  export let property
+
+  let selectedProperty
+  let searchType
+  let propertyImages
+  let themeClasses = {}
+
+  const unsubscribeSearchWidgetStore = searchWidgetStore.subscribe(store => {
+    selectedProperty = store.selectedProperty
+    searchType = store.searchType
+    propertyImages = store.propertyImages
+    themeClasses = store.themeClasses
+  })
+
+  const {
+    primaryHeading,
+    secondaryHeading,
+    primaryStrapline,
+    secondaryStrapline,
+    bodyText,
+    selectedItem,
+    resultItem,
+    offerBanner,
+  } = themeClasses
+
+  const id = (property && property.id) || ''
+  const propertyImage = propertyImages && propertyImages[id]
+  const imageUrl = (propertyImage && propertyImage.url) || INVALID_BACKGROUND_AS_BASE64
+  const sellingStatus = (property.selling && property.selling.status) || ''
+  const lettingStatus = (property.letting && property.letting.status) || ''
+
+  $: isSelectedProperty = property.id === (selectedProperty && selectedProperty.id) || ''
+
+  const selectProperty = () => {
+    searchWidgetStore.update(store => ({
+      ...store,
+      selectedProperty: property,
+    }))
+  }
+
+  const handleImageError = source => {
+    source.src = INVALID_BACKGROUND_AS_BASE64
+    source.target.src = INVALID_BACKGROUND_AS_BASE64
+    source.onerror = ''
+    return true
+  }
+
+  const dispatch = createEventDispatcher()
+
+  const handleBack = () => {
+    dispatch('back')
+  }
+
+  onMount(() => {
+    document.body.scrollTop = 0
+    document.documentElement.scrollTop = 0
+  })
+
+  onDestroy(() => {
+    unsubscribeSearchWidgetStore()
+  })
+</script>
+
+<style>
+  .property-item {
```

Comment:
Can we follow the component-name convention for classes eg `property-detail-item` makes conflicts less likely